src/processor/snapshot/importer.rs

```diff
@@ -44,8 +48,49 @@ impl SnapshotImporter {
 
     fn read_header(&self) -> Result<SnapshotHeader> {
         let header_path = self.directory.join(SNAPSHOT_HEADER_FILE_NAME);
-        let header_string = fs::read_to_string(header_path)?;
-        let header: SnapshotHeader = serde_json::from_str(&header_string)?;
+
+        let header = if let Ok(string) = fs::read_to_string(header_path) {
+            serde_json::from_str(&string)?
+        } else {
+            let mut l1_batch_number = U64::from(0);
+            let mut storage_logs_chunks = Vec::new();
+            let mut factory_deps_filepath = String::new();
+
+            let mut chunk_id = 0;
+            for file in self.directory.read_dir()? {
+                let file_name = file?
+                    .file_name()
+                    .to_str()
+                    .expect("invalid filename")
+                    .to_string();
+
+                if file_name.contains("storage_logs_part") {
```

Comment:
this probably works, but wouldn't regular expressions be easier?

Reply:
Perhaps, I'm still not fully convinced if this (generating our own header) is the best way to go about this. I'll spend some more time on this, but yes, regex would probably be a better fit.